opentracing-spring-cloud/src/test/java/io/opentracing/contrib/spring/cloud/jms/JmsTestConfiguration.java

```diff
@@ -0,0 +1,81 @@
+package io.opentracing.contrib.spring.cloud.jms;
+
+import java.util.HashSet;
+import java.util.logging.Logger;
+
+import javax.jms.ConnectionFactory;
+import javax.jms.Message;
+
+import org.apache.activemq.artemis.api.core.TransportConfiguration;
+import org.apache.activemq.artemis.core.config.impl.ConfigurationImpl;
+import org.apache.activemq.artemis.core.remoting.impl.invm.InVMAcceptorFactory;
+import org.apache.activemq.artemis.core.remoting.impl.netty.NettyAcceptorFactory;
+import org.apache.activemq.artemis.core.server.ActiveMQServer;
+import org.apache.activemq.artemis.core.server.impl.ActiveMQServerImpl;
+import org.apache.activemq.artemis.jms.client.ActiveMQJMSConnectionFactory;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.jms.annotation.JmsListener;
+
+/**
+ * @author <a href="mailto:ales.justin@jboss.org">Ales Justin</a>
+ */
+@Configuration
+public class JmsTestConfiguration {
```

Comment:
See at https://github.com/opentracing-contrib/java-spring-cloud/issues/14 I don't know what is the optimal solution. Modules allow to run all tests with one command, profiles are nice because does not introduce a lot of boilerplate but requires matrix builds.